### ISSUE TITLE
static when training on version 0.2

### ISSUE THREAD

**Issue by loganhart420** [coqui-ai/TTS#761]
trying to run the glow tts recipe  for ljspeech on version 0.2 produces static and just noise in the spectrograms. I have ran into this same issue trying to train a tacotron2 recipe as well and trying to train glow tts on other datasets. I am stumped on what is causing the problem, I even cloned a fresh repo to see if it was just my local fork and the issue persists. below is my tensorbaord and config. I'm not sure if this is an actual big or if I'm doing something wrong, any guidance is greatly appreciated 
**To Reproduce**
1. python recipes/ljspeech/glow_tts/train_glowtts.py 
2. run for 5000 steps 

**Environment (please complete the following information):**
Ubuntu 20.04 LTS
Pytorch 1.9.0 
Python 3.8
CUDA 11.1
RTX 3090 

config:
{
    "model": "glow_tts",
    "run_name": "",
    "run_description": "",
    "epochs": 1000,
    "batch_size": 128,
    "eval_batch_size": 64,
    "mixed_precision": false,
    "scheduler_after_epoch": false,
    "run_eval": true,
    "test_delay_epochs": -1,
    "print_eval": true,
    "dashboard_logger": "tensorboard",
    "print_step": 25,
    "plot_step": 100,
    "model_param_stats": false,
    "project_name": null,
    "log_model_step": null,
    "wandb_entity": null,
    "save_step": 10000,
    "checkpoint": true,
    "keep_all_best": false,
    "keep_after": 10000,
    "num_loader_workers": 4,
    "num_eval_loader_workers": 4,
    "use_noise_augment": false,
    "output_path": "../TTS/recipes/ljspeech/glow_tts",
    "distributed_backend": "nccl",
    "distributed_url": "tcp://localhost:54321",
    "audio": {
        "fft_size": 1024,
        "win_length": 1024,
        "hop_length": 256,
        "frame_shift_ms": null,
        "frame_length_ms": null,
        "stft_pad_mode": "reflect",
        "sample_rate": 22050,
        "resample": false,
        "preemphasis": 0.0,
        "ref_level_db": 20,
        "do_sound_norm": false,
        "log_func": "np.log10",
        "do_trim_silence": true,
        "trim_db": 45,
        "power": 1.5,
        "griffin_lim_iters": 60,
        "num_mels": 80,
        "mel_fmin": 0.0,
        "mel_fmax": null,
        "spec_gain": 20,
        "do_amp_to_db_linear": true,
        "do_amp_to_db_mel": true,
        "signal_norm": true,
        "min_level_db": -100,
        "symmetric_norm": true,
        "max_norm": 4.0,
        "clip_norm": true,
        "stats_path": null
    },
    "use_phonemes": false,
    "use_espeak_phonemes": true,
    "phoneme_language": "en-us",
    "compute_input_seq_cache": false,
    "text_cleaner": "english_cleaners",
    "enable_eos_bos_chars": false,
    "test_sentences_file": "",
    "phoneme_cache_path": "../TTS/recipes/ljspeech/glow_tts/phoneme_cache",
    "characters": {
        "pad": "_",
        "eos": "~",
        "bos": "^",
        "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!'(),-.:;? ",
        "punctuations": "!'(),-.:;? ",
        "phonemes": "iy\u0268\u0289\u026fu\u026a\u028f\u028ae\u00f8\u0258\u0259\u0275\u0264o\u025b\u0153\u025c\u025e\u028c\u0254\u00e6\u0250a\u0276\u0251\u0252\u1d7b\u0298\u0253\u01c0\u0257\u01c3\u0284\u01c2\u0260\u01c1\u029bpbtd\u0288\u0256c\u025fk\u0261q\u0262\u0294\u0274\u014b\u0272\u0273n\u0271m\u0299r\u0280\u2c71\u027e\u027d\u0278\u03b2fv\u03b8\u00f0sz\u0283\u0292\u0282\u0290\u00e7\u029dx\u0263\u03c7\u0281\u0127\u0295h\u0266\u026c\u026e\u028b\u0279\u027bj\u0270l\u026d\u028e\u029f\u02c8\u02cc\u02d0\u02d1\u028dw\u0265\u029c\u02a2\u02a1\u0255\u0291\u027a\u0267\u02b2\u025a\u02de\u026b"
    },
    "batch_group_size": 0,
    "loss_masking": null,
    "min_seq_len": 3,
    "max_seq_len": 500,
    "compute_f0": false,
    "compute_linear_spec": false,
    "add_blank": false,
    "datasets": [
        {
            "name": "ljspeech",
            "path": "../LJSpeech-1.1/",
            "meta_file_train": "metadata.csv",
            "ununsed_speakers": null,
            "meta_file_val": "",
            "meta_file_attn_mask": ""
        }
    ],
    "optimizer": "RAdam",
    "optimizer_params": {
        "betas": [
            0.9,
            0.998
        ],
        "weight_decay": 1e-06
    },
    "lr_scheduler": "NoamLR",
    "lr_scheduler_params": {
        "warmup_steps": 4000
    },
    "test_sentences": [],
    "use_speaker_embedding": false,
    "use_d_vector_file": false,
    "d_vector_dim": 0,
    "num_chars": null,
    "encoder_type": "rel_pos_transformer",
    "encoder_params": {
        "kernel_size": 3,
        "dropout_p": 0.1,
        "num_layers": 6,
        "num_heads": 2,
        "hidden_channels_ffn": 768,
        "input_length": null
    },
    "use_encoder_prenet": true,
    "hidden_channels_enc": 192,
    "hidden_channels_dec": 192,
    "hidden_channels_dp": 256,
    "dropout_p_dp": 0.1,
    "dropout_p_dec": 0.05,
    "mean_only": true,
    "out_channels": 80,
    "num_flow_blocks_dec": 12,
    "inference_noise_scale": 0.0,
    "kernel_size_dec": 5,
    "dilation_rate": 1,
    "num_block_layers": 4,
    "num_speakers": 0,
    "c_in_channels": 0,
    "num_splits": 4,
    "num_squeeze": 2,
    "sigmoid_scale": false,
    "data_dep_init_steps": 10,
    "style_wav_for_test": null,
    "length_scale": 1.0,
    "d_vector_file": false,
    "grad_clip": 5.0,
    "lr": 0.001,
    "r": 1
}
![Screenshot from 2021-08-23 01-58-49](https://user-images.githubusercontent.com/72301874/130398925-88cc001f-b748-4340-bc68-2a3fd880c420.png)
![Screenshot from 2021-08-23 01-58-56](https://user-images.githubusercontent.com/72301874/130398927-2c5be68d-efa3-4c00-ba62-aa5221cc9e06.png)

**Comment by erogol** [comment]
5k is quite early to stop training.

Do you also have `train` figures?

**Comment by loganhart420** [comment]
![Screenshot from 2021-08-23 02-15-34](https://user-images.githubusercontent.com/72301874/130445323-7c0a8342-33ee-41fb-a03d-d50d527c714e.png)
I had run it previously for 10k steps and still got the same output in terms of spectrograms, I just ran this run to show what the spectrograms looked liked because before I thought I was doing something wrong and just deleted that run folder, I can continue this current run again to sure though

**Comment by Sadam1195** [comment]
@loganhart420 I had similar experience with GlowTTS and after many changes in config it didn't work so I assumed something was broken ultimately I went with tacotron2.

**Comment by loganhart420** [comment]
Glad to know i’m not the only one having this issue, since no one else had reported on this I thought I was just going crazy haha, something else to note that running the same config does work on version 0.0.15.1, if this is an actual bug though I think it would be wise to try and find and fix it instead of just using tacotron2. i’m running two more trainings rn with more steps just to confirm including fine tuning a glow tts. i’m curious is version 0.2 the only version you have had this problem?

**Comment by loganhart420** [comment]
to add more context to the what is going on here on two runs, The first one is the glow tts recipe for ljspeech in the recipe folder at 22000 steps, the second is a finetuned vctk glow tts with same config and on vctk for 10k steps

ljspeech figs:
![Screenshot from 2021-08-23 21-20-29](https://user-images.githubusercontent.com/72301874/130540265-eb92011b-2694-46c9-be23-e5ec096a99e3.png)
![Screenshot from 2021-08-23 21-20-32](https://user-images.githubusercontent.com/72301874/130540266-30e3b225-9192-465a-8d98-4a63c55c041a.png)
![Screenshot from 2021-08-23 21-20-35](https://user-images.githubusercontent.com/72301874/130540268-f736c615-0a9f-4f6e-9d48-983ae16e0fbd.png)
![Screenshot from 2021-08-23 21-20-42](https://user-images.githubusercontent.com/72301874/130540270-f7c86660-5254-40b7-9acc-78374ab5e96d.png)

vctk fine tune:
![Screenshot from 2021-08-23 21-21-13](https://user-images.githubusercontent.com/72301874/130540314-b83d7de4-d4d9-4fec-a1b2-ff57d515728b.png)
![Screenshot from 2021-08-23 21-21-16](https://user-images.githubusercontent.com/72301874/130540316-97dbd4d6-f2c6-40ec-9981-d94300a17835.png)
![Screenshot from 2021-08-23 21-21-18](https://user-images.githubusercontent.com/72301874/130540317-82cb6d26-2cc0-4c9e-92bc-7d504bf104d5.png)
![Screenshot from 2021-08-23 21-21-22](https://user-images.githubusercontent.com/72301874/130540318-f892d085-9984-4327-b8e1-2b6968741a94.png)

**Comment by Sadam1195** [comment]
> Glad to know i’m not the only one having this issue, since no one else had reported on this I thought I was just going crazy haha, something else to note that running the same config does work on version 0.0.15.1, if this is an actual bug though I think it would be wise to try and find and fix it instead of just using tacotron2. i’m running two more trainings rn with more steps just to confirm including fine tuning a glow tts. i’m curious is version 0.2 the only version you have had this problem?


Somethings (restoring model, model loss values) seem broken after the major changes since 0.1.0 like ModelAPI, ModelManagerAPI, TrainerAPI. I tried using previous checkpoints but they seem incompatible with newer version releases and no layers are restored so had to train models from scratch. Yes, I could have tried to fix the code but since more changes are upcoming and I wasn't sure if spending time would worth it.

**Comment by Sadam1195** [comment]
> to add more context to the what is going on here on two runs, The first one is the glow tts recipe for ljspeech in the recipe folder at 22000 steps, the second is a finetuned vctk glow tts with same config and on vctk for 10k steps
> 

they both produce noise, right?

**Comment by loganhart420** [comment]
yea I didn’t include the audios because they are quite loud but it is the same noise that you get with any other run

**Comment by loganhart420** [comment]
> > Glad to know i’m not the only one having this issue, since no one else had reported on this I thought I was just going crazy haha, something else to note that running the same config does work on version 0.0.15.1, if this is an actual bug though I think it would be wise to try and find and fix it instead of just using tacotron2. i’m running two more trainings rn with more steps just to confirm including fine tuning a glow tts. i’m curious is version 0.2 the only version you have had this problem?
> 
> Somethings (restoring model, model loss values) seem broken after the major changes since 0.1.0 like ModelAPI, ModelManagerAPI, TrainerAPI. I tried using previous checkpoints but they seem incompatible with newer version releases and no layers are restored so had to train models from scratch. Yes, I could have tried to fix the code but since more changes are upcoming and I wasn't sure if spending time would worth it.

yea, that is true I haven't tried training any other tts model yet but did you only have this problem with glow tts since you said you went to tacotron2?

**Comment by Sadam1195** [comment]
I had similar problem with training HiFiGAN.
@loganhart420

**Comment by Sadam1195** [comment]
Did you fix this on your end?
@loganhart420

**Comment by loganhart420** [comment]
it was fixed with the newest version

**Comment by loganhart420** [comment]
> Did you fix this on your end?
> @loganhart420

It was fixed with the newest version update
